### PR TITLE
chore(x/intent): cleanup dead code

### PR DESCRIPTION
**Pull request by pitasi**
Remove ActionType, ActionHandler, and cleanup a couple functions.

## Summary by CodeRabbit

- **Refactor**
  - Simplified the `ApproveAction` method to directly execute actions without separate readiness checks.
  - Renamed and refactored methods to better reflect their functionalities in the action and intent handling processes.
  - Integrated functionality of `ParamKeyTable()` into `NewParams()` for streamlined parameter management.
  
- **Chores**
  - Updated `go.mod` to include `github.com/iancoleman/strcase` v0.3.0 as a direct dependency.
  
- **Removed**
  - Removed `GetActionMessage` function and `ParamKeyTable()` function, integrating their functionalities elsewhere.

### CODE REVIEW

File: proto/warden/intent/tx.proto [hunk 1]
```diff
@@ -71,7 +71,6 @@ message MsgNewActionResponse {
 message MsgApproveAction {
   option (cosmos.msg.v1.signer) = "creator";
   string creator     = 1;
-  string action_type = 2;
   uint64 action_id   = 3;
```
Comment (artur-abliazimov) [inline proto/warden/intent/tx.proto]: Should the `action_id` equals to 2 instead? Like, `uint64 action_id   = 2;`. The same question for the `MsgRevokeAction `.

**Comment by artur-abliazimov** [comment]
Do we use these keys anywhere?

https://github.com/warden-protocol/wardenprotocol/blob/1afd42af26a86325934803677d04075637376097/warden/x/intent/types/keys.go#L11C2-L11C13

https://github.com/warden-protocol/wardenprotocol/blob/1afd42af26a86325934803677d04075637376097/warden/x/intent/types/keys.go#L14

https://github.com/warden-protocol/wardenprotocol/blob/1afd42af26a86325934803677d04075637376097/warden/x/intent/types/keys.go#L17

**Comment by artur-abliazimov** [comment]
These functions are not used anywhere. Do we need them?

Functions:
https://github.com/warden-protocol/wardenprotocol/blob/1afd42af26a86325934803677d04075637376097/warden/x/intent/types/action.go#L45

https://github.com/warden-protocol/wardenprotocol/blob/1afd42af26a86325934803677d04075637376097/warden/x/intent/types/params.go#L10

Errors:
https://github.com/warden-protocol/wardenprotocol/blob/1afd42af26a86325934803677d04075637376097/warden/x/intent/types/errors.go#L12C2-L12C11

File: proto/warden/intent/tx.proto (hunk 1 above)
Comment (pitasi) [reply proto/warden/intent/tx.proto]: yes makes sense, before mainnet I aim to make this protobufs as clean as possible. After mainnet, we'll be much more careful than that and won't reuse tags 😬 

fixed

**Comment by pitasi** [comment]
> These functions are not used anywhere. Do we need them?
> 
> Functions:
> 
> https://github.com/warden-protocol/wardenprotocol/blob/1afd42af26a86325934803677d04075637376097/warden/x/intent/types/action.go#L45
> 
> https://github.com/warden-protocol/wardenprotocol/blob/1afd42af26a86325934803677d04075637376097/warden/x/intent/types/params.go#L10

thanks, removed!

> Errors: https://github.com/warden-protocol/wardenprotocol/blob/1afd42af26a86325934803677d04075637376097/warden/x/intent/types/errors.go#L12C2-L12C11

I'll leave this as part of #353

**Pull request merged**